### DISCUSSION TITLE
RAG over dependency code and docs

### DISCUSSION THREAD

**Discussion by AriMKatz** [community/community#87391] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

Can or will copilot business do RAG over code and docs of dependencies, or is this only an enterprise thing (or neither?). It's very hard to find specifics like this. The comparison matrix is pretty vague. 

This is an important feature that cursor provides. Thanks!

**Comment by ayim** [comment, marked as answer]
https://github.com/features/copilot

Per the comparison on the webpage, it seems that chat that's personalized to your code and allows doc search/summary will be a Copilot Enterprise feature as of Feb 2024.

![image](https://github.com/community/community/assets/10951869/c36ebd9f-53f8-4d7d-8757-90b49e19b82a)

**Reply by AriMKatz** [reply]
Thanks. What's the difference between "chat personalized to your codebase" and "code referencing'?

**Reply by ayim** [reply]
I think code referencing means you can specifically mention your code file when asking a question.
Chat personalized to your codebase means that the chat indexed your all your code files and docs and such to provide a more holistic and specific answer.

**Reply by phongtintruong** [reply]
Do you know how they index project code files and docs to retrieve only the relevant parts when prompting the LLM, the context length of a prompt is short while the context can be very long.
